### PR TITLE
Kiwigrid develop

### PR DESCRIPTION
**Pull request by krasv**
added mysql as persistency

### CODE REVIEW

File: ArgusCore/pom.xml [hunk 1]
```diff
@@ -381,6 +381,11 @@
             <artifactId>weka-stable</artifactId>
             <version>3.6.14</version>
         </dependency>
+		<dependency>
```
Comment (rmelick) [inline ArgusCore/pom.xml:384]: You can revert this, we're going to use postgres

File: ArgusCore/src/main/java/com/salesforce/dva/argus/entity/GlobalInterlock.java [hunk 2]
```diff
@@ -167,8 +167,9 @@ public static String obtainLock(EntityManager em, long expiration, long type, St
             tx.begin();
 
             GlobalInterlock lock = em.merge(new GlobalInterlock(type, note));
-
-            em.flush();
+            // just for mysql: @PrePersist does not work with EclipseLink and MySQL
+			lock.lockTime = System.currentTimeMillis();
```
Comment (rmelick) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/entity/GlobalInterlock.java:171]: revert this as well

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/DefaultAlertService.java [hunk 3]
```diff
@@ -290,6 +290,10 @@ public Alert findAlertByPrimaryKey(BigInteger id) {
 		List<History> historyList = new ArrayList<>();
 		List<AlertIdWithTimestamp> alertIdWithTimestampList = _mqService.dequeue(ALERT.getQueueName(), AlertIdWithTimestamp.class, timeout,
 				alertCount);
+		if(alertIdWithTimestampList.isEmpty()) {
+			_logger.info("no alerts due");
```
Comment (rmelick) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/alert/DefaultAlertService.java:294]: revert this, I made a separate commit (https://github.com/salesforce/Argus/pull/344/files) that has a PR against the public repo and is already merged to kiwigrid-develop

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/mq/kafka/Producer.java [hunk 4]
```diff
@@ -136,6 +136,7 @@ public Thread newThread(Runnable r) {
      */
     public <T extends Serializable> int enqueue(final String topic, List<T> objects) {
         int messagesBuffered = 0;
+        _logger.trace("enqueueing topic {}, {} objects", topic, objects.size());
```
Comment (rmelick) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/mq/kafka/Producer.java:139]: If you still want this, we should make a separate branch and PR against the public repo, then merge that branch into kiwigrid-develop

File: ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/DefaultTSDBService.java [hunk 5]
```diff
@@ -377,7 +377,17 @@ private ObjectMapper getMapper() {
                 chunkEnd = Math.min(objects.size(), chunkStart + CHUNK_SIZE);
                 try {
                     StringEntity entity = new StringEntity(fromEntity(objects.subList(chunkStart, chunkEnd)));
-                    HttpResponse response = executeHttpRequest(method, endpoint, entity);
+						_logger.debug("{} ({}) -> {}", endpoint, method, new Object() {
```
Comment (rmelick) [inline ArgusCore/src/main/java/com/salesforce/dva/argus/service/tsdb/DefaultTSDBService.java:380]: This is really weird looking, and I think your formatting is wrong.  There is a setting in IntelliJ (Editor -> Code Style : Indents Detection.  That will let you use the existing indents without having to change away from the kiwigrid style.

File: ArgusCore/src/test/java/com/salesforce/dva/argus/service/metric/transform/HoltWintersTransformTest.java [hunk 6]
```diff
@@ -31,15 +31,31 @@
...
 import java.util.HashMap;
+import java.util.Locale;
 import java.util.Map;
 
 import static org.junit.Assert.*;
 
 public class HoltWintersTransformTest {
 
-    @Test
+	private static Locale locale;
```
Comment (rmelick) [inline ArgusCore/src/test/java/com/salesforce/dva/argus/service/metric/transform/HoltWintersTransformTest.java:45]: Revert this, it has been handled in a separate PR that has already been merged: https://github.com/salesforce/Argus/pull/337

File: ArgusDocker/persistent/conf/argus-web-services/tomcat-users.xml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+	-->
+	<!--
+	  <role rolename="tomcat"/>
+	  <role rolename="role1"/>
+	  <user username="tomcat" password="<must-be-changed>" roles="tomcat"/>
+	  <user username="both" password="<must-be-changed>" roles="tomcat,role1"/>
+	  <user username="role1" password="<must-be-changed>" roles="role1"/>
+	-->
+	<role rolename="manager-gui"/>
+	<user username="argus" password="g3h31m" roles="manager-gui"/>
```
Comment (rmelick) [inline ArgusDocker/persistent/conf/argus-web-services/tomcat-users.xml:42]: I added these tomcat changes to the public docker PR and merged them in to kiwigrid-develop

File: ArgusDocker/persistent/conf/argus.properties [hunk 8]
```diff
@@ -1,15 +1,3 @@
-
-system.property.jpa.javax.persistence.schema-generation.database.action=create
-system.property.jpa.javax.persistence.jdbc.driver=org.postgresql.Driver
-system.property.jpa.javax.persistence.jdbc.url=jdbc:postgresql://postgres:5432/argus_user
-system.property.jpa.javax.persistence.jdbc.user=argus_user
-system.property.jpa.javax.persistence.jdbc.password=password
-system.property.jpa.eclipselink.ddl-generation=drop-and-create-tables
-system.property.jpa.eclipselink.logging.level=SEVERE
```
Comment (rmelick) [inline ArgusDocker/persistent/conf/argus.properties:8]: We need to keep these

File: ArgusDocker/simple/argus.properties [hunk 9]
```diff
@@ -1,5 +1,10 @@
...
+#build.property.persistence.unit=<provider>org.eclipse.persistence.jpa.PersistenceProvider</provider>\n#<exclude-unlisted-classes>false</exclude-unlisted-classes>\n    <properties>\n      <property name="javax.persistence.schema-generation.database.action" value="drop-and-create-tables"/>\n      <property name="javax.persistence.jdbc.driver" value="org.apache.derby.jdbc.EmbeddedDriver"/>\n      <property name="javax.persistence.jdbc.url" value="jdbc:derby:memory:argus;create=true"/>\n      <property name="javax.persistence.jdbc.user" value="root"/>\n      <property name="javax.persistence.jdbc.password" value="root"/>\n      <property name="eclipselink.ddl-generation" value="drop-and-create-tables"/>\n      <property name="eclipselink.logging.level" value="SEVERE"/>\n      <property name="eclipselink.logging.parameters" value="true"/>\n      <property name="eclipselink.target-database" value="DERBY"/>\n      <property name="eclipselink.canonicalmodel.subpackage" value="unit"/>\n    </properties>
+build.property.persistence.driver=com.mysql.jdbc.Driver
```
Comment (rmelick) [inline ArgusDocker/simple/argus.properties:3]: We should revert this and use the method that has already been merged into kiwigrid-develop to set jpa properties

**Comment by krasv** [comment]
all changes as been adapted.

**Pull request closed without merging**